### PR TITLE
Update for newest version of tRPC (v10)

### PR DESCRIPTION
**Pull request by sproott**
`resolveHTTPResponse` has been moved to `@trpc/server/http`. Also, the way of defining a router has changed in v10, I updated that in the README and tests.

### CODE REVIEW

**Comment by vishalbalaji** [comment]
@sproott, I was testing out your fork when I noticed some type errors with the `createTRPCHandle` hook in `hooks.server.ts`. There is a screenshot attached below as well.

2022-11-05_22:12:27_scrot

The errors are as follows:

```
Property 'routeId' is missing in type 'RequestEvent<Partial<Record<string, string>>, string | null>' but required in type 'RequestEvent<Partial<Record<string, string>>>'.
```

```
Type '(event: RequestEvent<Partial<Record<string, string>>, string | null>, opts?: ResolveOptions | undefined) => MaybePromise<Response>' is not assignable to type '(event: RequestEvent<Partial<Record<string, string>>>, opts?: ResolveOptions | undefined) => MaybePromise<Response>'.
  Types of parameters 'event' and 'event' are incompatible.
    Property 'route' is missing in type 'RequestEvent<Partial<Record<string, string>>>' but required in type 'RequestEvent<Partial<Record<string, string>>, string | null>'.
```

I am looking into why it might be, even though I have no clue as to what they even mean. I just thought I would let you know anyway.

**Comment by sproott** [comment]
@vishalbalaji The `createTRPCHandle` function takes the types of `event` and `resolve` straight from the `@sveltejs/kit` package and the types match with what's passed to `handle`, so it should "just work" and be passable to the function as-is.

This looks like a package version mismatch or something. Are you sure your environment and package versions are set up properly?

I couldn't really reproduce this myself, partly because I couldn't get my forked version to install properly with pnpm (first time I'm doing anything like this, not sure what's the correct way).

**Comment by vishalbalaji** [comment]
@sproott, I figured out why this might be happening. I think this issue pertains to the <code>[1.0.0-next.534](https://github.com/sveltejs/kit/releases/tag/%40sveltejs%2Fkit%401.0.0-next.534)</code> release  of `@sveltejs/kit`, where there was a breaking change replacing `routeId` with `route.id` in the `RequestEvent` type along with some other breaking changes.

I believe that you were not able to reproduce it becuase you may have the version of `@sveltejs/kit` pinned, but I had it set to `next` during my testing. This issue is resolved when reverting to the previous version: <code>[1.0.0-next.533](https://github.com/sveltejs/kit/releases/tag/%40sveltejs%2Fkit%401.0.0-next.533)</code>

**Comment by sproott** [comment]
@vishalbalaji So there's no problem with this PR then?

**Comment by vishalbalaji** [comment]
@sproott, technically, yes. However, I also noticed a small issue with downgrading though. Downgrading to <code>[@sveltejs/kit@1.0.0-next.533](https://github.com/sveltejs/kit/releases/tag/%40sveltejs%2Fkit%401.0.0-next.533)</code> does remove the type errors on `event` and `resolve`, but introduces a type error on `router`, the screenshot for which is attached below.

2022-11-07_00:12:54_scrot

For some reason, the type of the `appRouter` from `trpc` does not match with the `AnyRouter` type in `createTRPCHandle`. But this is easily solved by typecasting `appRouter` to `AnyRouter` like so:

```
import type { AnyRouter } from '@trpc/server'
import { createContext, responseMeta, appRouter } from '$lib/trpcServer'

...

export const handle: Handle = async ({ event, resolve }) => {
	const response = await createTRPCHandle({
               ...
		router: appRouter as AnyRouter,
               ...
	})

	return response
}

```

I would suggest adding this to the docs.

**TL;DR**, the issues brought up by me are solved by:

1. Pinning `@svelte/kit` to version: `1.0.0-next.533`.
2. Type casting `appRouter` as `AnyRouter`.

For now, just updating the versions and README should suffice. However, these are only workarounds to these issues and the types in `createTRPCHandle` hook will need to be updated to accommodate the breaking changes from `@svelte/kit` in the long run. I will try to open a PR to either your repo or this one if it is merged when I find the time.

**Comment by jameslounds** [comment]
Since it's using the types from "sveltekit-next-515", the `routeId` -> `route.id` is causing an issue.
Works for me if i update the `package.json` to next - same as my project - and run.

Quicker an easier workaround is just to change the `package.json` in `node_modules/trpc/sveltekit`, but tests on CI server still fail

**Comment by icflorescu** [comment]
Thanks a lot, but this is no longer necessary.
tRPC-SvelteKit `v3.x.x` is out and it comes with a documentation website here: [icflorescu.github.io/trpc-sveltekit](https://icflorescu.github.io/trpc-sveltekit)
